sagemaker-debugger/build_your_own_container_with_debugger/debugger_byoc.ipynb

```diff
@@ -0,0 +1,904 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "!{sys.executable} -m pip install \"sagemaker>=2.0.0\" smdebug"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### [Optional Step] Restart the kernel to apply the update"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Note:** If you are using **Jupyter Notebook**, the previous cell automatically installs and updates the libraries. If you are using **JupyterLab**, you have to manually choose the \"Restart Kernel\" under the **Kernel** tab in the top menu bar.\n",
+    "\n",
+    "Check the SageMaker Python SDK version by running the following cell."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 329,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "'2.5.0'"
+      ]
+     },
+     "execution_count": 329,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "import sagemaker\n",
+    "sagemaker.__version__"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Step 2: Prepare a Dockerfile and register the Debugger hook to you training script\n",
+    "\n",
+    "You need to put your **Dockerfile** and training script (**tf_keras_resnet_byoc.py** in this case) in the **docker** folder. Specify the location of the training script in the **Dockerfile** script in the line for `COPY` and `ENV`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Prepare a Dockerfile\n",
+    "\n",
+    "The following cell prints the **Dockerfile** in the **docker** folder. You must install `sagemaker-training` and `smdebug` libraries to fully access the SageMaker Debugger features."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 330,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "FROM tensorflow/tensorflow:2.2.0rc2-py3-jupyter\n",
+      "\n",
+      "RUN pip install sagemaker-training\n",
+      "RUN pip install smdebug\n",
+      "\n",
+      "#=====Copies the training code inside the container=====#\n",
+      "COPY tf_keras_resnet_byoc.py /opt/ml/code/tf_keras_resnet_byoc.py\n",
+      "\n",
+      "#=====Defines train.py as script entrypoint=============#\n",
+      "ENV SAGEMAKER_PROGRAM tf_keras_resnet_byoc.py"
+     ]
+    }
+   ],
+   "source": [
+    "! cat docker/Dockerfile"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Prepare a training script\n",
+    "\n",
+    "The following cell prints an example training script, **tf_keras_resnet_byoc.py**. To register the Debugger hook, you need to use the Debugger client library `smdebug`. \n",
```

Comment:
maybe just point to the training script, rather than printing the content in the notebook

Reply:
Enabled scrolling to not let it span too long. I wanted to keep it here so that customers do not have to go back and forth to the script file. Adding the link to it as well.